without-webhooks/server/php-slim/index.php

```diff
@@ -60,55 +67,66 @@ function generateResponse($intent, $logger)
     case "requires_source":
       // Card was not properly authenticated, suggest a new payment method
       return [
-        error => "Your card was denied, please provide a new payment method"
+        'error' => "Your card was denied, please provide a new payment method"
       ];
     case "succeeded":
       // Payment is complete, authentication not required
       // To cancel the payment after capture you will need to issue a Refund (https://stripe.com/docs/api/refunds)
       $logger->info("💰 Payment received!");
       return ['clientSecret' => $intent->client_secret];
   }
+
+  throw new \Exception('unsupported status');
 }
 
 $app->get('/stripe-key', function (Request $request, Response $response, array $args) {
-    $pubKey = getenv('STRIPE_PUBLISHABLE_KEY');
-    return $response->withJson(['publishableKey' => $pubKey]);
+    $pubKey = $_ENV['STRIPE_PUBLISHABLE_KEY'];
+    $response
+        ->withHeader('Content-Type', 'application/json')
+        ->getBody()
+        ->write(json_encode(['publishableKey' => $pubKey]))
+    ;
+    return $response;
 });
 
 
 $app->post('/pay', function(Request $request, Response $response) use ($app)  {
   $logger = $this->get('logger');
-  $body = json_decode($request->getBody());
+  $body = json_decode($request->getBody(), true);
   try {
-    if($body->paymentMethodId != null) {
+    if($body['paymentMethodId'] != null) {
       // Create new PaymentIntent with a PaymentMethod ID from the client.
       $intent = \Stripe\PaymentIntent::create([
-        "amount" => calculateOrderAmount($body->items),
-        "currency" => $body->currency,
-        "payment_method" => $body->paymentMethodId,
+        "amount" => calculateOrderAmount($body['items']),
+        "currency" => $body['currency'],
+        "payment_method" => $body['paymentMethodId'],
         "confirmation_method" => "manual",
         "confirm" => true,
+        // 3D Payment is not supported by this script, but the return_url is required in many cases.
+        "return_url"=> "http://localhost:4242/",
```

Comment:
This is the only change related to stripe. On my account, it was required to add this to make it work. I assume it's optional if your account is not set with 3DSecure only.

Anyway, it just changes basically nothing for most users, I can roll it back if you want. But I think keeping it is a good thing.